api/src/pcapi/core/subscription/api.py

```diff
@@ -734,7 +735,39 @@ def get_subscription_message(user: users_models.User) -> models.SubscriptionMess
     return None
 
 
+def _get_all_phone_validation_reason_codes(user: users_models.User) -> list[fraud_models.FraudReasonCode]:
+    return functools.reduce(
+        lambda acc, reason_codes: acc + reason_codes,
+        [
+            fraud_check.reasonCodes or []
+            for fraud_check in user.beneficiaryFraudChecks
+            if fraud_check.type == fraud_models.FraudCheckType.PHONE_VALIDATION
+        ],
+        [],
+    )
```

Comment:
(détail / question de goût) : Cette formulation (avec accumulateur + reduce) ne fait sûrement pas peur aux JavaScriptistes convaincus. En Python, j'utiliserais [itertools.chain.from_iterable](https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable) : 

```suggestion
    return itertools.chain.from_iterable(
        fraud_check.reasonCodes or []
        for fraud_check in user.beneficiaryFraudChecks
        if fraud_check.type == fraud_models.FraudCheckType.PHONE_VALIDATION
    )
```

Reply:
merci 👌 

on voulait faire un `flatten` ou `flapMap` mais on avait pas trouvé l'équivalent en python dans la lib standard